Makefile

```
@@ -0,0 +1,33 @@
+# Copyright (C) 2018-2020 Ruilin Peng (Nick) <pymumu@gmail.com>.
+#
+# smartdns is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# smartdns is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.PHONY: all
+
+all: SMARTDNS_BIN
+
+SMARTDNS_BIN : 
+	make -C src all $(MAKEFLAGS)
```

When using [recursive make](https://www.gnu.org/software/make/manual/html_node/Recursion.html), you should use MAKE variable:

```suggestion
	$(MAKE) -C src all $(MAKEFLAGS)
```

Though note recursive make is deemed problematic by some (e.g. https://embeddedartistry.com/blog/2017/04/10/recursive-make-considered-harmful/). But for a small project you can probably get away with it.